CRM/Dedupe/Finder.php

```diff
@@ -281,6 +281,8 @@ public static function formatParams($fields, $ctype) {
       if (preg_match('/(.*)-(Primary-[\d+])$|(.*)-(\d+|Primary)$/', $key, $matches)) {
         $return = array_values(array_filter($matches));
         $flat[$return[1]] = $value;
+        // make sure the first occurrence is kept, not the last
+        $flat[$return[1]] = !isset($flat[$return[1]]) ? $value : $flat[$return[1]];
```

Comment:
@jitendrapurohit what happens if the first is empty & the second isn't?

Reply:
Thanks @eileenmcnaughton. On testing the above scenario, I noticed it was still breaking if only the second field has the value in it. Fixed it and also extended the unit test to assert this additional behavior.